### PR TITLE
Key Management and Config File for Validator Onboarding

### PR DESCRIPTION
**Pull request by dailinsubjam**
This PR:
- Added a struct for ValidatorConfigFile and ValidatorConfig.
- Added a flow from reading config seed of file to generating a node's own key pairs.
- Make sure key generation only appear in test initialization or config initialization.
- Added a script to generate key pair. The steps are in https://github.com/EspressoSystems/HotShot/issues/1971.

Designs:
- The config file for validators contains two parts: seed & node_id. They together will server as a final seed for key generation. They can be used either for input of scripts for key generation or validator's config for test. The reason that it's not directly private key & public key is that the form of the keys are very complex (see`config/ValidatorConfigExample` for reference), it would be better to generate it rather than read it from config file, and also our `HotshotConfig` does contain the secret key originally for signning, so it's fine.
- An parameter added for gen_launcher like `gen_launcher::<TestTypes, MemoryImpl>(0)` is to identify `node_id` for key pair (pk & sk) generation, since `build_system_handle()` in `crates/testing/src/task_helpers.rs` needs launcher with different node id.

Main review parts: `orchestrator/src/config.rs`, `crates/types/src/lib.rs`.

### CODE REVIEW

**Review by bfish713 — DISMISSED** [review]
Nice, LGTM!

**Review by bfish713 — APPROVED** [review]

File: crates/hotshot-signature-key/src/bn254/bn254_priv.rs [hunk 1]
```diff
@@ -54,12 +54,9 @@ impl BLSPrivKey {
     }
 }
 
-#[allow(clippy::incorrect_partial_ord_impl_on_ord_type)]
```
Comment (rob-maron) [inline crates/hotshot-signature-key/src/bn254/bn254_priv.rs:57]: I'm good with this as long as it still works as intended ! I wonder why we implemented it this way to begin with

File: crates/hotshot/examples/infra/mod.rs [hunk 2]
```diff
@@ -94,38 +94,14 @@ pub struct OrchestratorArgs {
...
-    TYPES::SignatureKey,
-    <TYPES::SignatureKey as SignatureKey>::StakeTableEntry,
-    TYPES::ElectionConfigType,
-> {
+) -> NetworkConfig<TYPES::SignatureKey, TYPES::ElectionConfigType> {
     let config_file_as_string: String = fs::read_to_string(config_file.as_str())
         .unwrap_or_else(|_| panic!("Could not read config file located at {config_file}"));
-    let config_toml: NetworkConfigFile =
-        toml::from_str::<NetworkConfigFile>(&config_file_as_string)
+    let config_toml: NetworkConfigFile<TYPES::SignatureKey> =
```
Comment (rob-maron) [inline crates/hotshot/examples/infra/mod.rs:100]: This is clean! Love it

**Review by rob-maron — APPROVED** [review]
Looks good to me !

File: crates/hotshot-signature-key/src/bn254/bn254_priv.rs (hunk 1 above)
Comment (dailinsubjam) [reply crates/hotshot-signature-key/src/bn254/bn254_priv.rs:57]: I'm also wondering about that. In the beginning the implementation on this was very long ago, and later implementation just follow that. There are more discussions on whether we want to keep this [on zullip](https://espresso.zulipchat.com/#narrow/stream/311645-Engineering/topic/Clippy.20incorrect.20partial_cmp).

**Pull request merged**